### PR TITLE
Change runtime metrics tests to use tally TestScope

### PR DESCRIPTION
**Pull request by madhuravi**
Verify recorded metrics

### CODE REVIEW

**Comment by madhuravi** [comment]
@uber-go/service-framework

File: metrics/runtime_metrics_test.go [hunk 1]
```diff
@@ -75,3 +79,22 @@ func closeCollector(t *testing.T, r *RuntimeCollector) {
...
+	snapshot := scope.Snapshot()
+	// Check gauges
+	gauges := snapshot.Gauges()
+	assert.NotNil(t, gauges["num-goroutines"].Value())
+	assert.NotNil(t, gauges["gomaxprocs"].Value())
+	assert.NotNil(t, gauges["memory.allocated"].Value())
+	assert.NotNil(t, gauges["memory.heap"].Value())
+	assert.NotNil(t, gauges["memory.heapidle"].Value())
+	assert.NotNil(t, gauges["memory.heapinuse"].Value())
+	assert.NotNil(t, gauges["memory.stack"].Value)
```
Comment (glibsm) [inline metrics/runtime_metrics_test.go]: `Value()`

File: metrics/runtime_metrics_test.go [hunk 2]
```diff
@@ -75,3 +79,22 @@ func closeCollector(t *testing.T, r *RuntimeCollector) {
...
+	// Check counters
+	counters := snapshot.Counters()
+	assert.NotZero(t, counters["memory.num-gc"].Value())
+	// Check timers
+	timers := snapshot.Timers()
+	assert.NotZero(t, len(timers["memory.gc-pause-ms"].Values()))
```
Comment (glibsm) [inline metrics/runtime_metrics_test.go]: how does NotZero work with arrays? Is there `notEmpty` or something?

Comment (madhuravi) [inline metrics/runtime_metrics_test.go]: good point, fixed

**Pull request merged**